projects/evm-compatible-bridge/federator.md

```diff
@@ -0,0 +1,245 @@
+- Height
+  - Height of the first block that confirms the transaction.
+  - Transactions in Hathor do not have a height but we can use the `first_block_height` metadata.
+
+With this we can safely wait for 5 blocks on Hathor to have passed so we can create the proposal.
+There is an event `best-block-update` which tells the current height of the network.
+This proposal will be available for the polling API as soon as it is created.
+
+### API
+
+#### GET /events
+
+This is the polling api, an event is a request to collect signatures, this means there are events to collect signatures in the MultiSig wallet and events to collect signatures on the federation contract.
+
+Both kinds of events will be returned by this endpoint, the federator will be responsible to make the appropriate calls to the coordinator service or the federation contract.
+
+The response will contain a list of events and pagination data.
+
+```ts
+type Event = {
+  eventOrigin: 'htr' | 'evm' | 'admin',
+  type: 'mint' | 'melt' | 'send', // This indicates the action on the Hathor MultiSig wallet
+  proposalId: number,
+  txHex: string,
+  // The event will have 1 of the following fields
+  evmRequestId?: number,
+  htrRequestId?: number,
+  adminRequestId?: number,
+};
+
+type ApiResponse = {
+  events: Event[],
+  hasMore: boolean,
+};
+```
+
+#### POST /sign/{proposal_id}
+
+The proposal id is referring a Hathor tx proposal where the federator has signed and this api will be used to collect the signatures.
+
+Once enough signatures are collected, the transaction will be sent to the network.
+
+The endpoint will be expecting the fields:
+
+```ts
+type RequestBody = {
+  signature: string,
+  publicKey: string,
+};
+```
+
+The public key is required so we can count how many distinct signatures were collected.
+
+#### POST /request/evm
+
+The federators will call this endpoint when an event is received in the EVM loop.
+
+The endpoint will be expecting the fields:
+
+```ts
+type RequestBody = {
+  blockHash: string,
+  txHash: string,
+  evmTokenAddr: address,
+  amount: string,
+  destinationAddress: string,
+};
+```
+
+#### POST /fulfill/htr/{htr_request_id}
+
+When a request is fulfilled, the federator will call this endpoint to inform the API that the request was fulfilled.
+
+Hathor requests can only end when the tokens are sent or minted on the EVM chain.
+This means this endpoint will be called when the bridge contract has sent the event to confirm that the tokens were minted or unlocked.
+
+```ts
+type RequestBody = {
+  // Data to identify the transaction that fulfilled the request
+  blockHash: string,
+  txHash: string,
+  logIndex: number, // Of the event that confirmed the fulfillment
+};
+```
+
+#### POST /fulfill/evm/{evm_request_id}
+
+Requests originated in the EVM chain can only end when the Hathor request is sent to the network.
+This means this endpoint will be called when the coordinator has sent the transaction to the network.
```

Comment:
Since the coordinator is the one who sends the tx, does he need someone calling this API?

Reply:
You are correct, we can remove this api

Reply:
Was it done?

Reply:
Yes